### DISCUSSION TITLE
OpenFileDialog for UnPlatform MVVM C# Application running on Linux

### DISCUSSION THREAD

**Discussion by HowardPWeiss** [unoplatform/uno#12171] [Q&A]
I am creating a WPF MVVM application to run on both Windows and Linux.

Is there an equivalent of the Windows OpenFileDialog for a C# WPF MVVM application for Linux and, if so, how to access that FileDialog from a button in my main page?

Thanks

Howard Weiss

**Comment by jeromelaban** [comment]
To open files, see [Storage Pickers](https://platform.uno/docs/articles/features/windows-storage-pickers.html?q=storage) in our documentation.

**Reply by HowardPWeiss** [reply]
I do not appear to have access to the Windows.Storage namespace in my Skia.GTK project. GTK has a FilePicker. I am looking for an example of how to use GTK# or GTK with Visual Studio Code, UnoPlatform MVVM application written in C# and compiled with sorbet

**Reply by HowardPWeiss** [reply]
Compiled with dormer

**Reply by HowardPWeiss** [reply]
Dotnet

**Reply by jeromelaban** [reply]
You should be able to [use this exact code](https://platform.uno/docs/articles/features/windows-storage-pickers.html?q=storage#fileopenpicker). Which errors are you getting?

**Comment by HowardPWeiss** [comment]
Jerome

Some progress.  I added "using Windows.Storage" and "using Windows.Storage.Pickers" to my C# View Model and then added the following code, per the example which you provided

var fileOpenPicker = new FileOpenPicker();
fileOpenPicker.SuggestedStartLocation = PickerLocationId.ComputerFolder;
fileOpenPicker.FileTypeFilter.Add(".txt");
fileOpenPicker.FileTypeFilter.Add(".csv");
// For Uno.WinUI-based apps
var hwnd = WinRT.Interop.WindowNative.GetWindowHandle(App.MainWindow);
WinRT.Interop.InitializeWithWindow.Initialize(fileOpenPicker, hwnd);
StorageFile pickedFile = await fileOpenPicker.PickSingleFileAsync();

when I run dotnet build, it reports the following error

error CS4033: The 'await' operator can only be used within an async method. Consider marking this method with the 'async' modifier and changing its return type to 'Task'

Working on it

Howard Weiss

**Comment by HowardPWeiss** [comment]
Fixed it.  I had to declare the calling routine as async.  Thanks
